Added tests for the KeyboardHelper

mariusc opened the pull request:
This is my first try at testing delegates. Let me know if you think anything is not right and we can go over it

dominik-hadl commented on KeyboardHelperTests/Tests/KeyboardHelperTests.swift:
```diff
@@ -9,6 +9,93 @@
 import XCTest
 @testable import KeyboardHelper
 
+class ShowSpyDelegate : KeyboardNotificationDelegate {
```
Not a super fan of the names :D Could have been something more like `KeyboardShowingObject` and `KeyboardHidingObject`, but on the other hand, it's not a big issue.

mariusc replied:
Naming is hard... I called them like that because it's basically the test spy pattern http://xunitpatterns.com/Test%20Spy.html

dominik-hadl replied:
I guess you're right. It just felt disconnected from the code to me, but leave it as it is, it's fine :)

dominik-hadl commented on KeyboardHelperTests/Tests/KeyboardHelperTests.swift:
```diff
@@ -9,6 +9,93 @@
+        let expectation = expectationWithDescription("KeyboardHelper calls the delegate as the result of receiving the show notification")
+        spyDelegate.expectation = expectation
+
+        NSNotificationCenter.defaultCenter().postNotificationName(UIKeyboardWillShowNotification, object: kh)
```
It would be nice to send along custom `KeyboardApperanceInfo` and test it in the specific class (ShowSpy, HideSpy) for the same values, just so they don't get somehow changed by `NSNotificationCenter` or something :0

nickskull commented:
Added a few comments, other than that it looks fine :)

mariusc commented:
Added custom appearance info in the delegates test (although that was basically tested already in the KeyboardAppearanceInfoTests). Also added some more tests for KeyboardAppearanceInfoTests, for the  where the parsing fails and it uses the default values
